Index: package/kernel/mac80211/files/lib/wifi/mac80211.sh
===================================================================
--- package/kernel/mac80211/files/lib/wifi/mac80211.sh	(revision 43057)
+++ package/kernel/mac80211/files/lib/wifi/mac80211.sh	(working copy)
@@ -87,13 +87,7 @@
 
 		[ -n $htmode ] && append ht_capab "	option htmode	$htmode" "$N"
 
-		if [ -x /usr/bin/readlink ]; then
-			path="$(readlink -f /sys/class/ieee80211/${dev}/device)"
-			path="${path##/sys/devices/}"
-			dev_id="	option path	'$path'"
-		else
-			dev_id="	option macaddr	$(cat /sys/class/ieee80211/${dev}/macaddress)"
-		fi
+		dev_id="	option macaddr	$(cat /sys/class/ieee80211/${dev}/macaddress)"
 
 		cat <<EOF
 config wifi-device  radio$devidx
