Index: package/kernel/mac80211/files/lib/wifi/mac80211.sh
===================================================================
--- package/kernel/mac80211/files/lib/wifi/mac80211.sh	(revision 40431)
+++ package/kernel/mac80211/files/lib/wifi/mac80211.sh	(working copy)
@@ -623,7 +623,8 @@
 		}
 		iw phy "$dev" info | grep -q '2412 MHz' || { mode_band="a"; channel="36"; }
 
-		if [ -x /usr/bin/readlink ]; then
+		# CarrierWrt: We need the MAC address for other purposes, so let's stick with that for now.
+		if false && [ -x /usr/bin/readlink ]; then
 			path="$(readlink -f /sys/class/ieee80211/${dev}/device)"
 			path="${path##/sys/devices/}"
 			dev_id="	option path	'$path'"

