Index: package/base-files/files/sbin/sysupgrade
===================================================================
--- package/base-files/files/sbin/sysupgrade	(revision 43057)
+++ package/base-files/files/sbin/sysupgrade	(working copy)
@@ -9,6 +9,7 @@
 export INTERACTIVE=0
 export VERBOSE=1
 export SAVE_CONFIG=1
+export SAVE_CONFFILES=1
 export SAVE_OVERLAY=0
 export DELAY=
 export CONF_IMAGE=
@@ -27,7 +28,7 @@
 		-d) export DELAY="$2"; shift;;
 		-v) export VERBOSE="$(($VERBOSE + 1))";;
 		-q) export VERBOSE="$(($VERBOSE - 1))";;
-		-n) export SAVE_CONFIG=0;;
+		-n) export SAVE_CONFFILES=0;;
 		-c) export SAVE_OVERLAY=1;;
 		-b|--create-backup) export CONF_BACKUP="$2" NEED_IMAGE=1; shift;;
 		-r|--restore-backup) export CONF_RESTORE="$2" NEED_IMAGE=1; shift;;
@@ -103,7 +104,7 @@
 	( find $(sed -ne '/^[[:space:]]*$/d; /^#/d; p' \
 		/etc/sysupgrade.conf /lib/upgrade/keep.d/* 2>/dev/null) \
 		-type f 2>/dev/null;
-	  opkg list-changed-conffiles ) | sort -u > "$file"
+	  opkg list-changed-conffiles ) | sort -u >> "$file"
 	return 0
 }
 
@@ -114,21 +115,20 @@
 		-e '\,/META_[a-zA-Z0-9]*$,d' \
 		-e '\,/functions.sh$,d' \
 		-e '\,/[^/]*-opkg$,d' \
-	> "$file"
+	>> "$file"
 	return 0
 }
 
 # hooks
 sysupgrade_image_check="platform_check_image"
-[ $SAVE_OVERLAY = 0 -o ! -d /overlay/etc ] && \
-	sysupgrade_init_conffiles="add_uci_conffiles" || \
-	sysupgrade_init_conffiles="add_overlayfiles"
+sysupgrade_init_persistent=""
+sysupgrade_init_conffiles=""
 
 include /lib/upgrade
 
 [ "$1" = "nand" ] && nand_upgrade_stage2 $@
 
-do_save_conffiles() {
+do_save_config() {
 	local conf_tar="${1:-$CONF_TAR}"
 
 	[ -z "$(rootfs_type)" ] && {
@@ -136,7 +136,16 @@
 		ask_bool 0 "Abort" && exit
 		return 0
 	}
-	run_hooks "$CONFFILES" $sysupgrade_init_conffiles
+	touch $CONFFILES
+	if [ $SAVE_OVERLAY = 1 -a -d /overlay/etc ]; then
+		add_overlayfiles "$CONFFILES"
+	else
+		run_hooks "$CONFFILES" $sysupgrade_init_persistent
+		if [ $SAVE_CONFFILES = 1 -o $SAVE_OVERLAY = 1 ]; then
+			add_uci_conffiles "$CONFFILES"
+			run_hooks "$CONFFILES" $sysupgrade_init_conffiles
+		fi
+	fi
 	ask_bool 0 "Edit config file list" && vi "$CONFFILES"
 
 	v "Saving config files..."
@@ -154,7 +163,7 @@
 fi
 
 if [ -n "$CONF_BACKUP" ]; then
-	do_save_conffiles "$CONF_BACKUP"
+	do_save_config "$CONF_BACKUP"
 	exit $?
 fi
 
@@ -196,12 +205,10 @@
 		;;
 	esac
 	get_image "$CONF_IMAGE" "cat" > "$CONF_TAR"
-	export SAVE_CONFIG=1
-elif ask_bool $SAVE_CONFIG "Keep config files over reflash"; then
-	[ $TEST -eq 1 ] || do_save_conffiles
-	export SAVE_CONFIG=1
+elif ask_bool $SAVE_CONFFILES "Keep config files over reflash"; then
+	export SAVE_CONFFILES=1
 else
-	export SAVE_CONFIG=0
+	export SAVE_CONFFILES=0
 fi
 
 if [ $TEST -eq 1 ]; then
@@ -208,6 +215,8 @@
 	exit 0
 fi
 
+do_save_config
+
 run_hooks "" $sysupgrade_pre_upgrade
 
 ubus call system upgrade
